### PR TITLE
Add a test suite for TranslationServer

### PR DESCRIPTION
**Pull request by azuloo**
A test suit that covers `TranslationServer`, see #43440 for more details.

### CODE REVIEW

**Comment by azuloo** [comment]
I've fixed formatting issues and removed unneeded includes. Also squished 2 commits into one.

File: tests/core/string/test_translation_server.h [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+	Ref<Translation> t = memnew(Translation);
+	t->set_locale("uk");
+	t->add_message("Good Morning", String::utf8("Добрий ранок"));
+
+	TranslationServer *ts = TranslationServer::get_singleton();
+
+	int l_count_before = ts->get_loaded_locales().size();
+	ts->add_translation(t);
+	int l_count_after = ts->get_loaded_locales().size();
+	// Newly created Translation object should be added to the list, so the counter should increase, too
```
Comment (AThousandShips) [inline tests/core/string/test_translation_server.h:49]:
```suggestion
	// Newly created Translation object should be added to the list, so the counter should increase, too.
```

Comments should start with a capital and end with a period, same elsewhere

File: tests/core/string/test_translation_server.h [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+TEST_CASE("[TranslationServer] Locale operations") {
+	TranslationServer *ts = TranslationServer::get_singleton();
+
+	// Language variant test; we supplied the variant of Español and the result should be the same string
+	String loc = "es_Hani_ES_tradnl";
+	String res = ts->standardize_locale(loc);
+
+	CHECK(res == loc);
+
+	// No such variant in variant_map; should return everyting except the variant
```
Comment (AThousandShips) [inline tests/core/string/test_translation_server.h:74]:
```suggestion
	// No such variant in variant_map; should return everything except the variant
```

**Review by AThousandShips — CHANGES_REQUESTED** [review]

**Comment by AThousandShips** [comment]
Go ahead and update the commit message spelling when fixing the issues

**Comment by AThousandShips** [comment]
You accidentally reset your branch to `master`, automatically closing the PR, you need to open a new one

**Comment by Calinou** [comment]
Superseded by https://github.com/godotengine/godot/pull/79331.

**Pull request closed without merging**